x-pack/plugins/cloud_security_posture/server/saved_objects/migrations/csp_rule_template.ts

```diff
@@ -36,6 +37,20 @@ function migrateCspRuleMetadata(
   };
 }
 
+function deprecateCspRuleTemplateState(
+  doc: SavedObjectUnsanitizedDoc<CspRuleTemplateV840>,
+  context: SavedObjectMigrationContext
+): SavedObjectUnsanitizedDoc<CspRuleTemplateV870> {
+  const { ...metadata } = doc.attributes.metadata;
```

Comment:
```suggestion
  const { metadata } = doc.attributes;
```

or 

```suggestion
  const metadata = doc.attributes;
```

Reply:
can we add a comment similar to this:

// Keep only metadata, ignore state

i think this makes it easier to understand what is going on here.

Reply:
++. You could even make is explicit which attributes are removed with something like

```ts
const { muted, enabled, ...attributes } = doc.attributes;
return {
   ...obj,
   attributes
}
```

Reply:
Thanks @orouz and @pgayvallet for this comment - it really makes our code a lot more cleaner and maintainable. 

I'll go with @pgayvallet suggestion if that's ok :)